File: lab-2/src/moves/Slash.java
package moves;

import ru.ifmo.se.pokemon.*;

public class Slash extends PhysicalMove {
    public Slash(){
        super(Type.NORMAL, 70, 100);
    }

    protected void applySelfEffects(Pokemon p){
        p.setMod(Stat.SPEED, 4);
    }

    public String describe(){
        return "is using Slash";
    }
}

File: lab-2-kotlin/src/moves/Slash.kt
package moves

import ru.ifmo.se.pokemon.*

class Slash : PhysicalMove(Type.NORMAL, 70.0, 100.0) {
    override fun applySelfEffects(p: Pokemon) {
        p.setMod(Stat.SPEED, 4)
    }

    public override fun describe(): String {
        return "is using Slash"
    }
}